recipes/mosquitto/1.x/conanfile.py

```diff
@@ -0,0 +1,132 @@
+            self._cmake.configure(build_folder=self._build_subfolder)
+        return self._cmake
+
+    def _patch_sources(self):
+        if self.settings.os == "Windows":
+            if self.options.with_tls:
+                tools.replace_in_file(os.path.join(self._source_subfolder, "lib", "CMakeLists.txt"),
+                                    "set (LIBRARIES ${LIBRARIES} ws2_32)",
+                                    "set (LIBRARIES ${LIBRARIES} ws2_32 crypt32)")
+                tools.replace_in_file(os.path.join(self._source_subfolder, "src", "CMakeLists.txt"),
+                                    "set (MOSQ_LIBS ${MOSQ_LIBS} ws2_32)",
+                                    "set (MOSQ_LIBS ${MOSQ_LIBS} ws2_32 crypt32)")
+                tools.replace_in_file(os.path.join(self._source_subfolder, "src", "CMakeLists.txt"),
+                                    "target_link_libraries(mosquitto_passwd ${OPENSSL_LIBRARIES})",
+                                    "target_link_libraries(mosquitto_passwd ${OPENSSL_LIBRARIES} ws2_32 crypt32)")
+            tools.replace_in_file(os.path.join(self._source_subfolder, "lib", "CMakeLists.txt"),
+                                "install(TARGETS libmosquitto RUNTIME DESTINATION \"${CMAKE_INSTALL_BINDIR}\" LIBRARY DESTINATION \"${CMAKE_INSTALL_LIBDIR}\")",
+                                "install(TARGETS libmosquitto RUNTIME DESTINATION \"${CMAKE_INSTALL_BINDIR}\" LIBRARY DESTINATION \"${CMAKE_INSTALL_LIBDIR}\" ARCHIVE DESTINATION \"${CMAKE_INSTALL_LIBDIR}\")")
+            tools.replace_in_file(os.path.join(self._source_subfolder, "lib", "cpp", "CMakeLists.txt"),
+                                "install(TARGETS mosquittopp RUNTIME DESTINATION \"${CMAKE_INSTALL_BINDIR}\" LIBRARY DESTINATION \"${CMAKE_INSTALL_LIBDIR}\")",
+                                "install(TARGETS mosquittopp RUNTIME DESTINATION \"${CMAKE_INSTALL_BINDIR}\" LIBRARY DESTINATION \"${CMAKE_INSTALL_LIBDIR}\" ARCHIVE DESTINATION \"${CMAKE_INSTALL_LIBDIR}\")")
+
+    def build(self):
+        self._patch_sources()
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy(pattern="LICENSE.txt", dst="licenses", src=self._source_subfolder)
+        self.copy(pattern="edl-v10", dst="licenses", src=self._source_subfolder)
+        self.copy(pattern="epl-v10", dst="licenses", src=self._source_subfolder)
+        self.copy(pattern="mosquitto.conf", src=self._source_subfolder, dst="res")
+        cmake = self._configure_cmake()
+        cmake.install()
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+        tools.rmdir(os.path.join(self.package_folder, "etc"))
+        if not self.options.shared:
+            tools.remove_files_by_mask(os.path.join(self.package_folder, "lib"), "*.so*")
+            tools.remove_files_by_mask(os.path.join(self.package_folder, "lib"), "*.dll*")
```

Comment:
```suggestion
            tools.remove_files_by_mask(os.path.join(self.package_folder, "lib"), "*.dll*")
            tools.remove_files_by_mask(os.path.join(self.package_folder, "lib"), "*.dylib")
```

is not OSX included?

Reply:
Yes it is !

https://c3i.jfrog.io/c3i/misc/logs/pr/4288/1/mosquitto/1.6.12/a30d90046e1f9d72bf44f77be314fd5eb4494713/create_stdout.txt

-- Installing: /Users/jenkins/w/cci_PR-4288/.conan/data/mosquitto/1.6.12/_/_/package/a30d90046e1f9d72bf44f77be314fd5eb4494713/lib/libmosquitto.1.6.12.dylib
-- Installing: /Users/jenkins/w/cci_PR-4288/.conan/data/mosquitto/1.6.12/_/_/package/a30d90046e1f9d72bf44f77be314fd5eb4494713/lib/libmosquitto.1.dylib
-- Installing: /Users/jenkins/w/cci_PR-4288/.conan/data/mosquitto/1.6.12/_/_/package/a30d90046e1f9d72bf44f77be314fd5eb4494713/lib/libmosquitto.dylib